docs/source/develop.rst

```diff
@@ -78,28 +98,42 @@ The test suite contains three kinds of tests
     These are rare and mostly for testing the command line interface.
 
 If you are comfortable with the Tornado interface then you will be happiest
-using the ``@gen_cluster`` style of test
+using the ``@gen_cluster`` style of test, e.g.
 
 .. code-block:: python
 
-   from distributed.utils_test import gen_cluster
+    # tests/test_submit.py
 
-   @gen_cluster(client=True)
-   def test_submit(c, s, a, b):
-       assert isinstance(c, Client)
-       assert isinstance(s, Scheduler)
-       assert isinstance(a, Worker)
-       assert isinstance(b, Worker)
+    from distributed.utils_test import gen_cluster
+    from distributed.utils_test import inc
+    from distributed import Client
+    from distributed import Future
+    from distributed import Scheduler
+    from distributed import Worker
 
-       future = c.submit(inc, 1)
-       assert future.key in c.futures
 
-       # result = future.result()  # This synchronous API call would block
-       result = yield future
-       assert result == 2
+    @gen_cluster(client=True)
+    def test_submit(c, s, a, b):
+        assert isinstance(c, Client)
+        assert isinstance(s, Scheduler)
+        assert isinstance(a, Worker)
+        assert isinstance(b, Worker)
+
+        future = c.submit(inc, 1)
+        assert isinstance(future, Future)
+        assert future.key in c.futures
+
+        # result = future.result()  # This synchronous API call would block
+        result = yield future
+        assert result == 2
+
+        assert future.key in s.tasks
+        assert future.key in a.data or future.key in b.data
+
+
+.. code-block:: shell
```

Comment:
I think we can remove this code snippet as this section is generally about writing tests, not running them (that's described in the Dask contributing guide)